sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateSuite.scala

```diff
@@ -2358,4 +2382,50 @@ class TransformWithStateValidationSuite extends StateStoreMetricsTest {
     assert(ex1.getMessage.contains("Failed to find time values"))
     TransformWithStateVariableUtils.validateTimeMode(TimeMode.EventTime(), Some(10L))
   }
+
+  Seq(TimeMode.None(), TimeMode.ProcessingTime()).foreach { timeMode =>
+    test("transformWithState - using watermark but time mode as None should not perform " +
```

Comment:
nit: `time mode as None` is not the same with timeMode=$timeMode`

why not just `transformWithState - using watermark but time mode as $timeMode should not perform late record filtering` ?